lib/model/unreads.dart

```diff
@@ -125,6 +125,37 @@ class Unreads extends ChangeNotifier {
 
   int countInDmNarrow(DmNarrow narrow) => dms[narrow]?.length ?? 0;
 
+  int countInNarrow(Narrow narrow) {
+    switch (narrow) {
+      case DmNarrow():
+        return dms[narrow]?.length ?? 0;
+      case AllMessagesNarrow():
+        int c = 0;
+        for (final model in dms.values) {
+          c = c + model.length;
+        }
+        for (final streamData in streams.values) {
+          for (final model in streamData.values) {
+            c = c + model.length;
+          }
+        }
+        return c;
+      case StreamNarrow():
+        final streamData = streams[narrow.streamId];
+        if (streamData == null) return 0;
+        int c = 0;
+        for (final model in streamData.values) {
+          c = c + model.length;
+        }
+        return c;
+      case TopicNarrow():
+        final streamData = streams[narrow.streamId];
+        if (streamData == null) return 0;
+        final model = streamData[narrow.topic];
+        return model?.length ?? 0;
```

Comment:
Can use `?[]` similarly to `?.`:
```suggestion
        return streamData?[narrow.topic]?.length ?? 0;
```

(plus the renaming mentioned above)